src/main/java/randoop/reflection/OperationExtractor.java

```diff
@@ -118,18 +144,20 @@ public void visit(Method method) {
     }
     TypedClassOperation operation = TypedOperation.forMethod(method);
     if (classType.isSubtypeOf(operation.getDeclaringType()) && operation.isStatic()) {
+      // If this classType inherits this static method, but declaring class is not public, then
+      // consider method to have classType as declaring class.
       int declaringClassMods =
           method.getDeclaringClass().getModifiers() & Modifier.classModifiers();
       if (!Modifier.isPublic(declaringClassMods)) {
-        operation =
-            new TypedClassOperation(
-                operation.getOperation(),
-                classType,
-                operation.getInputTypes(),
-                operation.getOutputType());
+        operation = operation.getOperationForType(classType);
       }
     }
-    addOperation(operation);
+    if (operation == null) {
+      System.out.println("null operation for " + classType + " method " + method.getName());
```

Comment:
Should this code issue a more prominent error, such as crashing Randoop?
If not, make this message more specific about whether this is a problem or just normal operation.